x-pack/plugins/canvas/public/state/middleware/aeroelastic.js

```diff
@@ -56,16 +60,31 @@ const elementToShape = (element, i) => {
     aero.matrix.translate(cx, cy, z),
     aero.matrix.rotateZ(angleRadians)
   );
+  const isGroup = isGroupId(element.id);
+  const parent = (element.position && element.position.parent) || null; // reserved for hierarchical (tree shaped) grouping
   return {
     id: element.id,
-    parent: null, // reserved for hierarchical (tree shaped) grouping,
-    localTransformMatrix: transformMatrix,
+    type: isGroup ? 'group' : 'rectangleElement',
+    subtype: isGroup ? 'persistentGroup' : '',
+    parent,
     transformMatrix,
     a, // we currently specify half-width, half-height as it leads to
     b, // more regular math (like ellipsis radii rather than diameters)
   };
 };
 
+const shapeToElement = shape => {
+  return {
+    left: shape.transformMatrix[12] - shape.a,
```

Comment:
Either a comment or a constant that defines to what `12` and `13` refer... if I were fixing a bug in this file, I'd be scared to death to touch these lines.